base/support_menus/MainStatusStatistics.ts

```diff
@@ -0,0 +1,656 @@
+            let y_pos = MainStatusStatistics.ELEM.NUMBERS.Y;
+            let text = djinn_counts.set + "/" + djinn_counts.total;
+
+            if (this.state === MainStatusStates.CHARACTERS) {
+                y_pos += MainStatusStatistics.ELEM.DOWN_SHIFT;
+                text = String(djinn_counts.total);
+            }
+
+            let numbers = this.window.set_text_in_position(
+                text,
+                x_pos,
+                y_pos,
+                true,
+                false,
+                undefined,
+                false,
+                MainStatusStatistics.GROUP_KEY
+            );
+            this.state_sprites.push(numbers.text, numbers.shadow);
+
+            if (this.state !== MainStatusStates.CHARACTERS) {
+                y_pos += MainStatusStatistics.ELEM.NUMBERS.Y_SHIFT;
+                text = String(elemental_stats.level);
+
+                numbers = this.window.set_text_in_position(
+                    text,
+                    x_pos,
+                    y_pos,
+                    true,
+                    false,
+                    undefined,
+                    false,
+                    MainStatusStatistics.GROUP_KEY
+                );
+                this.state_sprites.push(numbers.text, numbers.shadow);
+
+                y_pos += MainStatusStatistics.ELEM.NUMBERS.Y_SHIFT;
+                text = String(elemental_stats.power);
+
+                numbers = this.window.set_text_in_position(
+                    text,
+                    x_pos,
+                    y_pos,
+                    true,
+                    false,
+                    undefined,
+                    false,
+                    MainStatusStatistics.GROUP_KEY
+                );
+                this.state_sprites.push(numbers.text, numbers.shadow);
+
+                y_pos += MainStatusStatistics.ELEM.NUMBERS.Y_SHIFT;
+                text = String(elemental_stats.resistance);
+
+                numbers = this.window.set_text_in_position(
+                    text,
+                    x_pos,
+                    y_pos,
+                    true,
+                    false,
+                    undefined,
+                    false,
+                    MainStatusStatistics.GROUP_KEY
+                );
+                this.state_sprites.push(numbers.text, numbers.shadow);
+            }
+
+            const djinni_elements = [elements.VENUS, elements.MERCURY, elements.MARS, elements.JUPITER];
+
+            for (let i = 0; i < djinni_elements.length; i++) {
+                const elem = djinni_elements[i];
+
+                const x_pos = MainStatusStatistics.ELEM.DJINN.CENTER_X + i * MainStatusStatistics.ELEM.DJINN.X_SHIFT;
+                let y_pos = MainStatusStatistics.ELEM.DJINN.Y;
+
+                if (this.state === MainStatusStates.CHARACTERS) y_pos += MainStatusStatistics.ELEM.DOWN_SHIFT;
+
+                const djinni_sprite = this.djinn_group.create(x_pos, y_pos, elem + "_djinn_set");
```

Comment:
This section seems to be repeating... 🤔
If it can't be reused, please use the same approach as before.